api/internal/messenger/database/contact_read.go

```diff
@@ -33,3 +34,58 @@ func (c *contactRead) Create(ctx context.Context, contactRead *entity.ContactRea
 	})
 	return err
 }
+
+func (c *contactRead) UpdateRead(ctx context.Context, params *UpdateContactReadFlagParams) error {
+	err := c.db.Transaction(ctx, func(tx *gorm.DB) error {
+		if _, err := c.getByContactIDAndUserID(ctx, params.ContactID, params.UserID); err != nil {
+			return err
+		}
+
+		updates := map[string]interface{}{
+			"read":       params.Read,
+			"updated_at": c.now(),
+		}
+
+		if params.UserID == "" {
+			err := tx.WithContext(ctx).
+				Table(contactReadTable).
+				Where("contact_id = ? AND user_id IS NULL", params.ContactID).
+				Updates(updates).Error
+			return err
+		}
+
+		err := tx.WithContext(ctx).
+			Table(contactReadTable).
+			Where("contact_id = ? AND user_id = ?", params.ContactID, params.UserID).
+			Updates(updates).Error
+		return err
```

Comment:
```go
/*
 * database.go
 */
type UpdateContactReadFlagParams struct {
	ContactID string
	UserID    *string
	Read      bool
}
```

paramsの定義を↑のようにstringのポインタで渡してあげることで綺麗にかけるかも..?

```suggestion
		err := tx.WithContext(ctx).
			Table(contactReadTable).
			Where("contact_id = ? AND user_id = ?", params.ContactID, params.UserID).
			Updates(updates).Error
		return err
```